### DISCUSSION TITLE
How to move/resize text or field?

### DISCUSSION THREAD

**Discussion by mmahalwy** [Hopding/pdf-lib#1213] [Q&A] [answered]
Curious, is it possible to move or resize fields or text? Similar to the options given in `addToPage`, would be create to move fields.

**Comment by Trapfether** [comment, marked as answer]
Hello, There is a lower level API available for setting the position of form elements.

Provided you have a reference to the PDFField you want to move / resize, the following code should work for you.

    const fieldWidget = field.getWidgets()[0]; // typically fields only have one widget, but exceptions exist
    const currentRectangle = fieldWidget.getRectangle(); // this is how you would get the current position / size
    fieldWidget.setRectangle({x: 0, y: 0, width: 100, height: 100}); // and here is the function you are looking for specifically
